### PR TITLE
Add SIP service

### PR DESCRIPTION
**Pull request by dennwc**

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]
lgtm!

File: src/SipClient.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  DeleteSIPDispatchRuleRequest,
+  SIPParticipantInfo,
+  CreateSIPParticipantRequest,
+} from '@livekit/protocol';
+import ServiceBase from './ServiceBase.js';
+import { livekitPackage, Rpc, TwirpRpc } from './TwirpRPC.js';
+import {
+  SIPDispatchRuleDirect,
+  SIPDispatchRuleIndividual,
+} from '@livekit/protocol/src/gen/livekit_sip_pb';
```
Comment (lukasIO) [inline src/SipClient.ts:21]: I think we should export this directly from the protocol package's index file, as this pattern might cause errors for some picky bundlers.

Comment (lukasIO) [reply src/SipClient.ts:21]: adding `SIPDispatchRuleDirect, SIPDispatchRuleIndividual` to the import from `@livekit/protocol` should work.

Comment (dennwc) [reply src/SipClient.ts]: Done!

**Comment by dennwc** [comment]
@lukasIO what would be the easiest way to test it? I'm not very familiar with JS ecosystem.

**Comment by lukasIO** [comment]
you could just add a `my-sip-test.ts` file in the root of your local repo and import (and use) the new APIs from there. 

You can run it directly with e.g. `pnpm dlx tsx ./my-sip-test.ts`

**Review by lukasIO — APPROVED** [review]

**Pull request merged**